app-creator/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -0,0 +1,226 @@
+    const reordering = (widgetWrapper as DraggableWidget).editable;
+    if (reordering) {
+      if (nextElement == null) {
+        container.appendChild(widgetWrapper);
+      } else {
+        container.insertBefore(widgetWrapper, nextElement);
+      }
+      return;
+    }
+
+    // making clone with new id
+    const clone = widget.cloneNode(true) as HTMLElement;
+    clone.id += `-${store.widgetIDs[widget.id]}`;
+
+    // check if the element already exists
+    // this is necessary because the event is fired multiple times consecutively
+    const element = this.shadowRoot?.getElementById(`${clone.id}-wrapper`);
+    if (element != null) {
+      container.removeChild(element);
+    }
+
+    // the cloned widget is not wrapped with a draggable widget so we have to create one below
+    const cloneDraggableWrapper = document.createElement('draggable-widget');
+    cloneDraggableWrapper.editable = true;
+    cloneDraggableWrapper.style.width = '100%';
+    cloneDraggableWrapper.id = `${clone.id}-wrapper`;
+    cloneDraggableWrapper.appendChild(clone);
+
+    if (nextElement == null) {
+      container.appendChild(cloneDraggableWrapper);
+    } else {
+      container.insertBefore(cloneDraggableWrapper, nextElement);
+    }
+
+    // we use this to correctly increment the widget id
+    store.elementAdded = true;
+
+    // we hide the placeholder if it exists
+    const placeholder = this.shadowRoot?.getElementById('empty-placeholder');
+    if (placeholder == null) {
+      return;
+    }
+
+    placeholder.style.display = 'none';
+  }
+
+  /**
+   * Returns the next closest widget to the one being dragged, or null if there are no widgets after.
+   * @param widget widget currently being dragged
+   * @param y the y coordinate of the triggered event
+   */
+  getNextElement(widget: HTMLElement, y: number): Element | null {
+    const container = this.shadowRoot?.getElementById('container');
+    if (container == null) {
+      return null;
+    }
+
+    const children = Array.from(container.children).filter(
+      (child) =>
+        child.id !== `${widget.id}-${store.widgetIDs[widget.id]}-wrapper`
+    );
+
+    const closest = children.reduce(
+      (closest: { diff: number; element: Element | null }, child) => {
+        const rect = child.getBoundingClientRect();
+        const diff = y - rect.top - rect.height / 2;
+        if (diff < 0 && diff > closest.diff) {
+          return { diff, element: child };
+        } else {
+          return closest;
+        }
+      },
+      {
+        diff: Number.NEGATIVE_INFINITY,
+        element: null,
+      }
+    );
+
+    return closest.element;
+  }
+
+  /**
+   * Highlights dropzone border on drag enter and modifies flex alignment.
+   */
+  handleDragenter(e: Event) {
+    // return early if dragenter is called on child widget
+    if ((e.target as HTMLElement).id !== 'container') {
```

Comment:
Since 'container' is used in multiple places, might be worth using a constant variable (like CONTAINER) for it?

Reply:
Something like this ?
```Typescript
class Dropzone extends LitElement {
  CONTAINER: HTMLElement | null = null;

  firstUpdated() {
    this.CONTAINER = this.shadowRoot?.getElementById('container') ?? null;
  }
}
```

usage 
```
instanceMethod() {
   const {CONTAINER} = this;
   if (CONTAINER == null) return; 
   // method body
}
```